### PR TITLE
Enable Content Generators

### PR DESCRIPTION
**Pull request by maxamillion**

### CODE REVIEW

File: osbs/build/spec.py [hunk 1]
```diff
@@ -258,6 +260,7 @@ def set_params(self, sources_command=None, architecture=None, vendor=None,
         self.koji_target.value = koji_target
         self.kojiroot.value = kojiroot
         self.kojihub.value = kojihub
+        self.koji_certs_secret = koji_certs_secret
```
Comment (twaugh) [inline osbs/build/spec.py]: Here's the corresponding bug we expected earlier. This needs to be

```
self.koji_certs_secret.value = koji_certs_secret
```

Comment (twaugh) [inline osbs/build/spec.py]: I just understood what you told me earlier - this comes from another PR. I've added comments there about the two instancesides of missing `.value`, and you'll be able to re-base once they're fixed.

In the meantime you could add the fixes to this pull request as a separate commit that will be easy to skip on re-base, just for the purpose of getting the Travis testing to pass.

**Comment by twaugh** [comment]
Now that https://github.com/projectatomic/osbs-client/pull/377 is merged, please re-base against master.

**Comment by twaugh** [comment]
Also:
* please squash together related commits
* the `or None` change (c038b02) is no longer needed.

**Comment by maxamillion** [comment]
Rebased and removed change from https://github.com/projectatomic/osbs-client/commit/c038b025d92e2a83a8ed1e8b1ca758cadf72ec09

File: osbs/build/build_request.py [hunk 2]
```diff
@@ -473,6 +484,21 @@ def adjust_for_triggers(self):
...
+        Remove koji Content Generator related plugins if no triggers set in
+        order to hadle the "scratch build" scenario
+        """
+        if self.scratch:
+            # Note: only one for now, but left in a list like other adjust_for_
+            # functions in the event that this needs to be expanded
+            for when, which in [
+                ("exit_plugins", "koji_promote"),
+            ]:
+                logger.info("removing %s from request because there are no triggers",
```
Comment (twaugh) [inline osbs/build/build_request.py]: This should be a message about scratch builds, not triggers, I think?

File: osbs/build/build_request.py [hunk 3]
```diff
@@ -748,6 +774,7 @@ def render(self, validate=True):
             del self.template['spec']['triggers']
 
         self.adjust_for_triggers()
+        self.adjust_for_scratch()
```
Comment (twaugh) [inline osbs/build/build_request.py]: If we ran `adjust_for_scratch()` before `adjust_for_triggers()`, we could actually remove any triggers for this build. We never want to have automated rebuilds for scratch builds.

(Note: this adjusts the BuildConfig, so kills any existing triggers... fixing that is separate issue, https://github.com/projectatomic/osbs-client/issues/370, which we won't worry about here)

**Comment by twaugh** [comment]
Did some changes get lost? This looks like all it does is deal with scratch builds.

**Comment by twaugh** [comment]
I think the missing parts are:
* don't remove koji_promote when there are no triggers
* don't run the cp_built_image_to_nfs plugin at all (`render_cp_built_image_to_nfs` can go, and the references to it in `inputs/*.json` can go as well)

**Comment by maxamillion** [comment]
I don't know, I rebased as requested and this is what I got ... let me inspect git history and figure out what happened.

**Comment by twaugh** [comment]
I added another commit on https://github.com/twaugh/osbs-client/tree/content-generator to fix up the tests, based on what you had before.

So what's left is: run adjust_for_scratch() before adjust_for_triggers(); fix adjust_for_scratch log message.

**Comment by twaugh** [comment]
Oh, and you'll need to cherry-pick that change I made (it's 7106f3a).

**Comment by maxamillion** [comment]
@twaugh - Thank you, I missed that ... been badly multi-tasking all morning which is effectively done nothing good for anyone. Apologies.

**Comment by twaugh** [comment]
The scratch/triggers message, and removing triggers for scratch builds, can be dealt with later.
